stlab/concurrency/actor.hpp

```diff
@@ -0,0 +1,179 @@
+/*
+    Copyright 2023 Adobe
+    Distributed under the Boost Software License, Version 1.0.
+    (See accompanying file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+*/
+
+/**************************************************************************************************/
+
+#pragma once
+#ifndef STLAB_CONCURRENCY_ACTOR_HPP
+#define STLAB_CONCURRENCY_ACTOR_HPP
+
+// stdc++
+#include <memory>
+#include <optional>
+
+// stlab
+#include <stlab/concurrency/serial_queue.hpp>
+#include <stlab/concurrency/set_current_thread_name.hpp>
+
+// lambda pack captures are a C++20 feature, but both clang and msvc support it, so we should be
+// okay to use it even under C++17.
+#if __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wc++20-extensions"
+#endif // __clang__
+
+/**************************************************************************************************/
+
+namespace stlab {
+inline namespace v1 {
+
+/**************************************************************************************************/
+
+namespace detail {
+
+/**************************************************************************************************/
+
+struct temp_thread_name {
+    explicit temp_thread_name(const char* name) { stlab::set_current_thread_name(name); }
```

Comment:
I would not set the name of the thread. It is not worth the "trouble" from my point of view. The tasks should be short living anyway so setting the name does not make really sense.  If it is just for some milliseconds who will observer the thread name?

Reply:
This is very helpful for debugging, as often you will break in a random thread (managed by the thread pool) and you will want to know specifically which actor you are dealing with (especially when there are many `actor<T>` instances floating around for the same `T`). I suppose a workaround would be to have some kind of identifier in the `T` of every actor, but that seemed like a less useful pattern than letting the actor name the thread while it is using it.